### PR TITLE
Use <For/> in `YearPicker` documentation 

### LINKED ISSUE #3730: Use `<For />` in `YearPicker` documentation
### Provide the exact quote of the error or issue
We should use `<For />` component when repeating a component
e.g.
```tsx
<For each={items}>{(item, index) => <Text key={index} fontWeight="bold">{item}</Text>}</For>
```

https://yamada-ui.com/components/forms/year-picker

### Does the docs page already exist? Please link to it.

_No response_

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### PR DESCRIPTION
**Pull request by kaehehehe**
Closes #3730

## Description

Fixed the repetition part in the `YearPicker` documentation to use the `For` component.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 1]
```diff
@@ -28,59 +28,95 @@ import { YearPicker } from "@yamada-ui/calendar"
...
 
 カレンダーのサイズを変更する場合は、`calendarSize`を設定します。
 
 ```tsx
 <VStack>
-  <YearPicker placeholder="small size" calendarSize="sm" />
-  <YearPicker placeholder="medium size" calendarSize="md" />
-  <YearPicker placeholder="large size" calendarSize="lg" />
+  <For
+    each={[
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:68]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 2]
```diff
@@ -28,59 +28,95 @@ import { YearPicker } from "@yamada-ui/calendar"
 
 ```tsx
 <VStack>
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:30]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 3]
```diff
@@ -28,59 +28,95 @@ import { YearPicker } from "@yamada-ui/calendar"
...
 </VStack>
 ```
 
 カレンダーのバリアントを変更する場合は、`variant`を設定します。
 
 ```tsx
 <VStack>
-  <YearPicker placeholder="solid" calendarVariant="solid" />
-  <YearPicker placeholder="subtle" calendarVariant="subtle" />
+  <For each={["solid", "subtle"]}>
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:43]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 4]
```diff
@@ -28,59 +28,95 @@ import { YearPicker } from "@yamada-ui/calendar"
...
 
 ### サイズを変更する
 
 ```tsx
 <VStack>
-  <YearPicker placeholder="extra small size" size="xs" />
-  <YearPicker placeholder="small size" size="sm" />
-  <YearPicker placeholder="medium size" size="md" />
-  <YearPicker placeholder="large size" size="lg" />
+  <For each={["solid", "subtle"]}>
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:55]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 5]
```diff
@@ -223,9 +259,16 @@ import "dayjs/locale/ja"
 
 ```tsx
 <VStack>
-  <YearPicker isDisabled variant="outline" placeholder="outline" />
-  <YearPicker isDisabled variant="filled" placeholder="filled" />
-  <YearPicker isDisabled variant="flushed" placeholder="flushed" />
+  <For each={["outline", "filled", "flushed"]}>
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:262]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 6]
```diff
@@ -235,9 +278,16 @@ import "dayjs/locale/ja"
 
 ```tsx
 <VStack>
-  <YearPicker isReadOnly variant="outline" placeholder="outline" />
-  <YearPicker isReadOnly variant="filled" placeholder="filled" />
-  <YearPicker isReadOnly variant="flushed" placeholder="flushed" />
+  <For each={["outline", "filled", "flushed"]}>
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:281]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.ja.mdx [hunk 7]
```diff
@@ -247,9 +297,16 @@ import "dayjs/locale/ja"
 
 ```tsx
 <VStack>
-  <YearPicker isInvalid variant="outline" placeholder="outline" />
-  <YearPicker isInvalid variant="filled" placeholder="filled" />
-  <YearPicker isInvalid variant="flushed" placeholder="flushed" />
+  <For each={["outline", "filled", "flushed"]}>
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.ja.mdx:300]: @kaehehehe 
It seems that using `<For />` has actually reduced readability. Why don't we try replacing `<For />` with something else?

File: docs/contents/components/forms/year-picker/index.mdx [hunk 8]
```diff
@@ -28,59 +28,95 @@ import { YearPicker } from "@yamada-ui/calendar"
```
Comment (koralle) [inline docs/contents/components/forms/year-picker/index.mdx:28]: @kaehehehe 
This is the same as `docs/contents/components/forms/year-picker/index.ja.mdx`.

**Review by koralle — COMMENTED** [review]
@kaehehehe 
Thank you for the PR.

Overall, I notice many places where using <For /> seems to reduce readability. 
There is one spot where I think it could go either way - using <For /> or not using it would both be fine. 
One option would be to close this PR altogether - what do you think?

File: docs/contents/components/forms/year-picker/index.ja.mdx (hunk 2 above)
Comment (kaehehehe) [reply docs/contents/components/forms/year-picker/index.ja.mdx:30]: @koralle 
I think it's better to use the `For` component. Why do you think the readability is lower?

File: docs/contents/components/forms/year-picker/index.ja.mdx (hunk 3 above)
Comment (kaehehehe) [reply docs/contents/components/forms/year-picker/index.ja.mdx:43]: @koralle 
I think it's better to use the For component. Why do you think the readability is lower?

File: docs/contents/components/forms/year-picker/index.ja.mdx (hunk 4 above)
Comment (kaehehehe) [reply docs/contents/components/forms/year-picker/index.ja.mdx:55]: I think it's better to use the For component. Why do you think the readability is lower?

File: docs/contents/components/forms/year-picker/index.ja.mdx (hunk 5 above)
Comment (kaehehehe) [reply docs/contents/components/forms/year-picker/index.ja.mdx:262]: I think it's better to use the For component. Why do you think the readability is lower?

File: docs/contents/components/forms/year-picker/index.ja.mdx (hunk 7 above)
Comment (kaehehehe) [reply docs/contents/components/forms/year-picker/index.ja.mdx:300]: I think it's better to use the For component. Why do you think the readability is lower?

**Comment by kaehehehe** [comment]
I have removed the `For` component in cases where using it made the code longer. Please rereview. @koralle

**Pull request closed without merging**